front/src/views/Boxes/components/BoxesActionsAndTable.tsx

```diff
@@ -38,54 +54,251 @@ function BoxesActionsAndTable({
     navigate(`/bases/${baseId}/boxes/${labelIdentifier}`);
 
   // --- Actions on selected Boxes
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
-  const [selectedBoxes, setSelectedBoxes] = useState<Row<any>[]>([]);
+  const [selectedBoxes, setSelectedBoxes] = useState<Row<BoxRow>[]>([]);
+  const thereIsABoxMarkedForShipmentSelected = useMemo(
+    () =>
+      selectedBoxes.some(
+        (box) => box.values.shipment !== null && box.values.state === BoxState.MarkedForShipment,
+      ),
+    [selectedBoxes],
+  );
+
   // Move Boxes
-  // TODO: replace following refetchQuery with something directly writing to the cache
-  const moveBoxesAction = useMoveBoxes([
-    {
-      query: BOXES_LOCATIONS_TAGS_SHIPMENTS_FOR_BASE_QUERY,
-      variables: {
-        baseId,
-      },
-    },
-  ]);
+  const moveBoxesAction = useMoveBoxes();
 
   const onMoveBoxes = useCallback(
-    (locationId: string) =>
-      moveBoxesAction.moveBoxes(
-        selectedBoxes.map((box) => (box as unknown as BoxRow).labelIdentifier),
+    async (locationId: string) => {
+      const newAlerts: IAlerts[] = [];
+      const movableLabelIdentifiers = selectedBoxes
+        .filter(
+          (box) =>
+            ![BoxState.Receiving, BoxState.MarkedForShipment, BoxState.InTransit].includes(
+              box.values.state,
+            ),
+        )
+        .map((box) => box.values.labelIdentifier);
+
+      const boxCountInShipmentStates = selectedBoxes.length - movableLabelIdentifiers.length;
+      if (boxCountInShipmentStates > 0) {
+        newAlerts.push({
+          id: alerts.length,
```

Comment:
@HaGuesto I need to check this more carefully, but I feel that in general all of these should be toasts and not alerts. The general rule of thumb is: toasts for dynamic responses to user actions; alerts for persistent states OR confirmation/denial that has detailed information the user needs to write down.

Reply:
I have some comments on the texts for the messages too. Would you like a new Trello card for this and the above feedback? @HaGuesto